.NET/Microsoft.Recognizers.Text.Number/Japanese/Parsers/JapaneseNumberParser.cs

```diff
@@ -0,0 +1,495 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Recognizers.Text.Number.Japanese
+{
+    public class JapaneseNumberParser : BaseNumberParser
+    {
+        protected new readonly JapaneseNumberParserConfiguration Config;
+
+        public JapaneseNumberParser(JapaneseNumberParserConfiguration config) : base(config)
+        {
+            this.Config = config;
+        }
+
+        public override ParseResult Parse(ExtractResult extResultJpn)
+        {
+            string extra = null;
+            ParseResult ret = null;
+            extra = extResultJpn.Data as string;
+
+            var getExtResultJpn = new ExtractResult()
+            {
+                Start = extResultJpn.Start,
+                Length = extResultJpn.Length,
+                Data = extResultJpn.Data,
+                Text = extResultJpn.Text,
+                Type = extResultJpn.Type
+            };
+
+
+            Console.WriteLine("extra: " + extra);
+
+
+            if (extra == null)
+            {
+                return null;
+            }
+
+            if (extra.Contains("Ordinal"))
+            {
+                ret = OrdParseJpn(getExtResultJpn);
+                Console.WriteLine("Ordinal: ");
+            }
+            else if (extra.Contains("Per"))
+            {
+                ret = PerParseJpn(getExtResultJpn);
+                Console.WriteLine("Per: ");
+            }
+
+            if (ret != null)
+            {
+                ret.Text = extResultJpn.Text;
+            }
+
+            return ret;
+        }
+
+        private string ReplaceFullWithHalf(string text)
```

Comment:
Are this and ReplaceTraWithSim common to Japanese and Chinese code? If so, shouldn't them be in some reusable class?
Also, can you add some code documentation explaining these methods?